### PR TITLE
TEALv4: Knowable creatable ID

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
     * Byteslice arithmetic ops (`b+`, `b-`, `b*`, `b/`, `b%`, `b<`, `b>`, `b<=`, `b>=`, `b==`, `b!=`, `b\`, `b&`, `b^`, `b~`, `bzero`).
     * Additional mathematical opcodes: `divmodw`, `exp`, `expw`, `shl`, `shr`, `sqrt`
 + Updated all examples & tests to use TEALv4 (`#pragma version 4`)
++ TEALv4: gaid, gaids opcodes
```
Comment (ratik21) [inline CHANGELOG.md:19]: Could you please add more description here?

File: packages/runtime/src/ctx.ts [hunk 2]
```diff
@@ -643,12 +646,13 @@ export class Ctx implements Context {
           };
           this.tx = this.gtxs[idx]; // update current tx to the requested index
 
-          this.addApp(
+          const appID = this.addApp(
             fromAccountAddr, flags,
             txnParam.approvalProgram,
-            txnParam.approvalProgram,
+            txnParam.clearProgram,
```
Comment (ratik21) [inline packages/runtime/src/ctx.ts:652]: why are we doing this change?

File: packages/runtime/src/interpreter/opcode-list.ts [hunk 3]
```diff
@@ -3249,3 +3249,61 @@ export class Sqrt extends Op {
...
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    super();
+    this.line = line;
+    this.interpreter = interpreter;
+    assertLen(args.length, 1, line);
+    this.txIndex = Number(args[0]);
+  };
+
+  execute (stack: TEALStack): void {
+    const ID = this.interpreter.runtime.ctx.knowableID.get(this.txIndex);
```
Comment (ratik21) [inline packages/runtime/src/interpreter/opcode-list.ts:3276]:
```suggestion
    const knowableID = this.interpreter.runtime.ctx.knowableID.get(this.txIndex);
```

File: packages/runtime/test/integration/knowable-id.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import { types } from "@algo-builder/web";
+import { assert } from "chai";
+
+import { RUNTIME_ERRORS } from "../../src/errors/errors-list";
+import { AccountStore, Runtime } from "../../src/index";
+import { AppDeploymentFlags } from "../../src/types";
```
Comment (ratik21) [inline packages/runtime/test/integration/knowable-id.ts:6]: linter is throwing warning here.

File: packages/runtime/src/ctx.ts (hunk 2 above)
Comment (amityadav0) [reply packages/runtime/src/ctx.ts:652]: it was a bug before

File: CHANGELOG.md (hunk 1 above)
Comment (amityadav0) [reply CHANGELOG.md:19]: ok

File: packages/runtime/test/integration/knowable-id.ts (hunk 4 above)
Comment (amityadav0) [reply packages/runtime/test/integration/knowable-id.ts:6]: ok

**Review by ratik21 — APPROVED** [review]

**Pull request merged**